CT-1571 parallelise tests

misaka opened the pull request.

EdwardAndress commented:
Why is an empty job and, corresponding, empty spec added here?

misaka commented:
Because I wanted to have some kind of job in here, even if it gets renamed later, etc. I was assuming that we would require _a_ job to be able to spin up SideKiq.

EdwardAndress commented:
Can you test that?

EdwardAndress commented:
I approve either way

EdwardAndress commented:
Although given that ActiveJob is also compatible with other queueing systems, it would be possible to switch out Sidekiq for something else at some point in the future.  This would be made easier if adding sidekiq were separated from adding the jobs, at least in terms of the commits used.

misaka commented:
@EdwardAndress I didn't think it was worth proving, tbh. I don't think there's enough benefit to be had, even if we do decide to rip out SideKiq I don't think we'd do a revert commit, I think we'd just add the gem and edit the config files.

EdwardAndress commented:
OK

misaka commented:
@aliuk2012 @EdwardAndress Sorry for more changes coming through, but there was some things that needed fixing. If you could re-review I'd appreciate it. :)

misaka committed:
CT-1571 parallelise tests

misaka committed:
CT-1571 merge parallel:create and ...:prepare into ...:setup

misaka committed:
CT-1571 try 8 processes in parallel

misaka committed:
CT-1571 remove non-parallel db loading

misaka committed:
CT-1571 move DB init to before_script for travis

misaka committed:
CT-1571 find_or_create case outcomes in tests

misaka committed:
CT-1571 create and load db structure, without seeding

misaka committed:
CT-1571 isolate feature tests to their own rspec process

In an attempt to get rid of errors we get when running tests in parallel on
Travis.

misaka committed:
CT-1571 fix typo: s/parallel_test/parallel_rspec

misaka committed:
CT-1571 parameterise the number of tests to run in parallel

misaka committed:
CT-1571 add support for uploading test failure screenshots to S3

Every now and then we have sporadic failures in Travis that much upset and
gnashing of teeth. This enables a mechanism built-into capybara-screenshots to
upload sceenshots of errors directly to S3 from the Travis host for our viewing
pleasure.

To enable this feature, env vars have to be set in Travis for the access key and
secret access key of a user that has access to a bucket we've set aside for this
purpose. File uploaded to this but will be expired after 8 weeks.

misaka committed:
CT-1571 Add information about rspec timing

By providing a log file with the amount of time tests take to run, the parallel
test runner is better-able to balance out the number of specs running in each
process. Useful for when running parallel tests in Travis.

misaka committed:
CT-1571 update README.md
